nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -41,6 +47,38 @@ in {
       description = "Whether to enable the systemd-boot (formerly gummiboot) EFI boot manager";
     };
 
+    signed = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Whether or not the bootloader files, including systemd-boot
+        EFI programs should be signed.
+      '';
+    };
+
+    signing-key = mkOption {
+      type = types.path;
+      example = "/root/secure-boot/db.key";
+      description = ''
+        The <literal>db.key</literal> signing key, for signing EFI
+        programs. Note: Do not pass a store path. Passing the key like
+        <literal>signing-key = ./db.key;</literal> will copy the
```

Comment:
Couldn't this hazard be avoided by using `toString`, i.e. `signingKey = if cfg.signed then (toString cfg.signing-key) else "/no-signing-key";`? Learned this from [this stackoverflow answer](https://stackoverflow.com/questions/43850371/when-does-a-nix-path-type-make-it-into-the-nix-store-and-when-not)